### ISSUE TITLE
Validating dictionaries

### ISSUE THREAD

**Issue by dennisk8** [thephpleague/openapi-psr7-validator#17]
Hello everyone,
from my understanding, a dictionary can set a type inside additionalProperties to define the type of values. So in the example below it should only allow string-string key-value pairs.

```json
{
  "type": "object",
  "additionalProperties": {
    "type": "string"
  }
}
```

However, the validator does not validate whether the properties are of the specified type. E.g. following validates without error even though it should:

```json
{ 
   "one":"1",
   "two":"2",
   "three":{ 
      "four":"4"
   }
}
```

From what I can tell from debugging is that it only does a type check on $data (if it's an object), but does not check the individual properties if they are the specified type defined in additionalProperties. Is it a bug/missing feature? I could also be very wrong so feel free to correct me.

P.s. thank you for all the hard work you have put into this tool, it's great!

**Comment by lezhnev74** [comment]
It can be be a thing actually. Can you provide a small failing test for this?

**Comment by dennisk8** [comment]
Hey, sorry for the slow reply. I'm actually really new at this, should I do a pull request, or? :')

**Comment by scaytrase** [comment]
PR would be absolutly better, but you can just provide a code snipped

**Comment by dennisk8** [comment]
@scaytrase sorry for the late response. Thanks a bunch for the fix, it works great!
I was testing it out and might have stumbled upon an issue which could be related to the fix. When the definition is defined as:

```json
{
  "myData": {
    "type": "object",
      "additionalProperties": {
        "type": "string"
      }
  }
}
```

While the validation works great for the elements inside the object, it seems to have a problem to validate the object itself. It will fail correctly when the value is e.g. a `string` or an `integer`, but if you pass an `array` it wrongly validates it as correct.

e.g. following example validates as true
```json
"myData": ["this", "should", "be", "invalid"]
```

When I do the following:
```json
"myData": {
  "first":"one",
  "second": {
    "third":"three"
  }
}
```
I receive the error `myData.second: Value expected to be 'string', 'array' given.`, even though it should say `'object' given`. Could it be that it may have a problem to distinguish between `array` and `object`?

**Comment by scaytrase** [comment]
Yeah, maybe

**Comment by scaytrase** [comment]
First one fixed in #24. For the second one I think we should utilize (object) `stdClass` instead of assoc array inside  the whole lib. This would be pretty large refactoring, but useful. As a workaround we could oveerride `gettype` to local type extractor which would respect assoc array as object, but not sure that it's worth it.
